cmake/winml_unittests.cmake

```diff
@@ -66,17 +66,19 @@ add_winml_test(
 )
 target_precompiled_header(winml_test_api testPch.h)
 
-file(GLOB winml_test_scenario_src CONFIGURE_DEPENDS "${WINML_TEST_SRC_DIR}/scenario/cppwinrt/*.cpp")
-add_winml_test(
-  TARGET winml_test_scenario
-  SOURCES ${winml_test_scenario_src}
-  LIBS winml_test_common onnxruntime_providers_dml
-  DEPENDS winml_api
-)
-target_precompiled_header(winml_test_scenario testPch.h)
-set_target_properties(winml_test_scenario PROPERTIES LINK_FLAGS
-  "/DELAYLOAD:d2d1.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll"
-)
+if (onnxruntime_USE_DML)
```

Comment:
>onnxruntime_USE_DML [](start = 4, length = 19)

just confirming - we won't run scenario tests without DML?

Reply:
My understanding is that it's not possible in its current state, right? Scenario tests don't compile without DML.

I'll change the tests to skip compilation of tests that depend on DML when it's disabled (add some `#ifdef`), but I think this PR could be merged as-is.

Reply:
sounds good

---
In reply to: [351031299](https://github.com/microsoft/onnxruntime/pull/2479#discussion_r351031299) [](ancestors = 351031299)

Reply:
I changed it yesterday to build scenario tests except CustomOps.cpp but forgot to push the changes, pushed them now.